src/test/java/org/jboss/xavier/integrations/EndToEndTest.java

```diff
@@ -0,0 +1,524 @@
+    private static Logger logger = LoggerFactory.getLogger(EndToEndTest.class);
+
+
+    @ClassRule
+    public static GenericContainer activemq = new GenericContainer<>("vromero/activemq-artemis")
+            .withExposedPorts(61616, 8161)
+            .withLogConsumer(new Slf4jLogConsumer(logger).withPrefix("AMQ-LOG"))
+            .withEnv("DISABLE_SECURITY", "true")
+            .withEnv("BROKER_CONFIG_GLOBAL_MAX_SIZE", "50000")
+            .withEnv("BROKER_CONFIG_MAX_SIZE_BYTES", "50000")
+            .withEnv("BROKER_CONFIG_MAX_DISK_USAGE", "100");
+
+    @ClassRule
+    public static GenericContainer kie_server = new GenericContainer<>("jboss/kie-server-showcase:7.18.0.Final")
+            .withNetworkAliases("kie-server")
+            .withExposedPorts(8080)
+            .withLogConsumer(new Slf4jLogConsumer(logger).withPrefix("KIE-LOG"))
+            .withEnv("KIE_SERVER_ID", "analytics-kieserver")
+            .withEnv("KIE_ADMIN_USER", "kieserver")
+            .withEnv("KIE_ADMIN_PWD", "kieserver1!")
+            .withEnv("KIE_SERVER_MODE", "DEVELOPMENT")
+            .withEnv("KIE_MAVEN_REPO", "https://oss.sonatype.org/content/repositories/snapshots")
+            .withEnv("KIE_REPOSITORY","https://repository.jboss.org/nexus/content/groups/public-jboss")
+            .withEnv("KIE_SERVER_CONTROLLER_PWD","admin")
+            .withEnv("KIE_SERVER_CONTROLLER_USER","admin")
+            .withEnv("KIE_SERVER_LOCATION","http://kie-server:8080/kie-server/services/rest/server")
+            .withEnv("KIE_SERVER_PWD","kieserver1!")
+            .withEnv("KIE_SERVER_USER","kieserver");
+
+    @ClassRule
+    public static PostgreSQLContainer postgreSQL = new PostgreSQLContainer()
+            .withDatabaseName("sampledb")
+            .withUsername("admin")
+            .withPassword("redhat");
+
+    @ClassRule
+    public static LocalStackContainer localstack = new LocalStackContainer()
+            .withLogConsumer(new Slf4jLogConsumer(logger).withPrefix("AWS-LOG"))
+            .withServices(S3);
+
+    private static String ingressCommitHash = "3ea33a8d793c2154f7cfa12057ca005c5f6031fa"; // 2019-11-11
+
+    @Inject
+    private InitialSavingsEstimationReportService initialSavingsEstimationReportService;
+
+    @Autowired
+    private InitialSavingsEstimationReportRepository initialSavingsEstimationReportRepository;
+
+    @Inject
+    private AnalysisRepository analysisRepository;
+
+    @Value("${S3_BUCKET}")
+    private String bucket;
+
+    @Value("${server.port:8080}")
+    private String serverPort;
+
+    @Value("${test.timeout.performance:60000}") // 1 minute
+    private int timeoutMilliseconds_PerformaceTest;
+
+    @Value("${test.timetout.ics:10000}") // 10 seconds
+    private int timeoutMilliseconds_InitialCostSavingsReport;
+
+    @Value("${minio.host:x}")
+    private String minio_host;
+
+    public static class Initializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
```

Comment:
Since we have the limitation of writing just one Test method by Class, then I would suggest moving the `Initializer` class outside the `EndToEndTest.java` class since it might be used by future Test Class. What do you think?

Reply:
As we discussed; my idea is to make some parts of the Class reusable by future End2End test classes. However, I agree that for now there are no other End2End tests in mind so we can postpone the refactoring of this class

Reply:
As we talked, moving the initializer out of the class will convert it in a non static class .... and also it needs the values from the containers started ( ip and port ) there fore we need to have the initializer and the test class on sync. So I dont see much benefit on moving the class out and it will complicate things a bit.
Also, I think reusable is interesting but only when you need to reuse....